### DISCUSSION TITLE
How to disable nuclei from scanning port not specified

### DISCUSSION THREAD

**Discussion by iuliu8899** [projectdiscovery/nuclei#5159] [Q&A]
I run nuclei by this command

`nuclei -u ip:8080`

My aim is to just scan 8080 port of this ip. But it still scanned other port.

![image](https://github.com/projectdiscovery/nuclei/assets/31680027/7248109d-bc44-438c-97cd-3725953eb29d)

It discover mysql on 3306, redis on 6379 and ssh on 22 port.

My question is: How to disable this feature? (since I only hope to scan specific port)

**Comment by rxerium** [comment]
Hey @iuliu8899, I'd suggest narrowing down the templates you are using to `http`, this can be done by using the following command:
```
nuclei -u ip:8080 -t ~/nuclei-templates/http
```

Replace `~/nuclei-templates/http` with the path to the `http` folder.

Hope this helps 😊

**Reply by iuliu8899** [reply]
Thanks for your reply. 

But what if I don't know if it's a http service on the port? 

I mean if the target.txt contains e.g. both http service and mysql service like: 

ip:8080
ip:3306

for this case, if I only use http pocs (by using `-t`), result will be not accurate.

**Reply by rxerium** [reply]
So to clarify, you'd like hosts with ports associated with http (80/443/8080/10443 etc..) to run against http templates and hosts with other ports to run against their associated services? Such as, a host with a port 21 will run against FTP templates?

**Reply by iuliu8899** [reply]
E.g. now I have a target.txt file, in which 2 target defined:
Ip:3306
Ip:8080

Then if I start nuclei scan by
'nuclei -l target.txt -t ~/nuclei-templates/http/'

The result will be not accurate since we only use http pocs.

But if I run by

'Nuclei -l target.txt'

The results will contains many ports not in my target.txt, such as redis 6379 if there is a redis service on 6379 port. how to disable that.

Hope I described clearly...

**Reply by rxerium** [reply]
Apologies for the delayed reply. As far as I know this is not possible however I will leave this issue for the PD team to confirm.

**Reply by GeorginaReeder** [reply]
Hey @iuliu8899 , we can take a look into this for you. Bear with us while I share with the team!

**Reply by iuliu8899** [reply]
@GeorginaReeder @rxerium 
thank you all!!!

**Reply by gunnsth** [reply]
This would be fantastic.  We might know what ports are open based on an nmap scan.  I guess if the templates could be filtered by port that would suffice?

**Comment by ehsandeep** [comment]
@iuliu8899 you can exclude tcp and javascript protocol based template that has hardcoded ports for certain service / exploit detection to avoid this.

`nuclei -ept tcp,javascript`

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
